lib/os/memcpy32.c

```diff
@@ -0,0 +1,49 @@
+#include <sys/util.h>
+#include <sys/__assert.h>
+#include <stdint.h>
+#include <toolchain.h>
+
+/*
+ * Copyright (c) 2021 Pete Dietl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define IS_ALIGNED(X) (!((uintptr_t)X & (sizeof(uint32_t) - 1)))
+
+void *memcpy32(void *restrict dst0, const void *restrict src0, size_t len)
```

Comment:
Strange, I thought restrict was only useful if the same memory address was read multiple times.

But we don't read the same address multiple times in this memcpy.

Did you see any changes to the generated machine code with and without the restrict keyword?

Reply:
Regular `memcpy` uses `restrict` in its parameters, so I did the same here.

Reply:
I removed it anyway

Reply:
In Zephyr, you can use `ZRESTRICT`.